sql/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseExpiryCheck.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.reference.sys.check.cluster;
+
+import io.crate.expression.reference.sys.check.SysCheck;
+import io.crate.license.DecryptedLicenseData;
+import io.crate.license.LicenseExpiryNotification;
+import io.crate.license.LicenseService;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.inject.Singleton;
+
+import java.util.concurrent.CompletableFuture;
+
+import static io.crate.expression.reference.sys.check.AbstractSysCheck.CLUSTER_CHECK_LINK_PATTERN;
+import static io.crate.expression.reference.sys.check.AbstractSysCheck.getLinkedDescription;
+import static io.crate.expression.reference.sys.check.SysCheck.Severity.HIGH;
+import static io.crate.expression.reference.sys.check.SysCheck.Severity.MEDIUM;
+import static io.crate.license.LicenseExpiryNotification.MODERATE;
+
+@Singleton
+public class LicenseExpiryCheck implements SysCheck {
+
+    private static final int ID = 5;
+    private static final BytesRef LICENSE_NOT_CLOSE_TO_EXPIRY_DESCRIPTION = new BytesRef("Your CrateDB is not close to expiry. Enjoy CrateDB!");
+
+    private BytesRef description = LICENSE_NOT_CLOSE_TO_EXPIRY_DESCRIPTION;
+    private Severity severity = Severity.LOW;
+    private final LicenseService licenseService;
+
+    @Inject
+    public LicenseExpiryCheck(LicenseService licenseService) {
+        this.licenseService = licenseService;
+    }
+
+    @Override
+    public boolean validate() {
+        DecryptedLicenseData currentLicense = licenseService.currentLicense();
```

Comment:
If enterprise is disabled, the check should probably always be good?

Reply:
With disabled license we'll still have an unlimited license so it would be valid by virtue of a never expiring license